rfc-177-edition-all-content.md

```diff
@@ -0,0 +1,148 @@
+
+    WorldwideOrganisationWorldLocation ||--|| WorldLocation : "has one"
+    Sponsorship ||--|| Organisation : "has one"
+    WorldwideOrganisationRole ||--|| Role : "has one"
+    EditionWorldwideOrganisation ||--|| WorldNewsStory : "has one"
+    EditionWorldwideOrganisation ||--|{ CorporateInformationPage : "has many"
+```
+
+The diagram below shows the relationship between models in Whitehall after Worldwide Organisations were made editionable:
+
+```mermaid
+---
+title: "Architecture of worldwide organisation model's relationship to other models after being made editionable in Whitehall"
+---
+
+erDiagram
+    Attachment {}
+    Document {}
+    EditionableWorldwideOrganisation["EditionableWorldwideOrganisation (Edition)"] {}
+    EditionEditionableWorldwideOrganisation {}
+    EditionOrganisation {}
+    EditionRole {}
+    EditionWorldLocation {}
+    FeaturedImageData {}
+    Organisation {}
+    Role {}
+    SocialMediaAccount {}
+    WorldLocation {}
+    WorldNewsStory["WorldNewsStory (Edition)"] {}
+    WorldwideOffice {}
+    WorldwideOrganisationPage {}
+
+    Document ||--|{ EditionableWorldwideOrganisation : "has many"
+
+    EditionableWorldwideOrganisation ||--|{ EditionWorldLocation : "has many"
+    EditionableWorldwideOrganisation ||--|{ SocialMediaAccount : "has many"
+    EditionableWorldwideOrganisation ||--|{ EditionOrganisation : "has many"
+    EditionableWorldwideOrganisation ||--|{ WorldwideOffice : "has many"
+    EditionableWorldwideOrganisation ||--|{ EditionRole : "has many"
+    EditionableWorldwideOrganisation ||--|{ EditionEditionableWorldwideOrganisation : "has many"
+    EditionableWorldwideOrganisation ||--|{ WorldwideOrganisationPage : "has many"
+    EditionableWorldwideOrganisation ||--|{ FeaturedImageData : "has many"
+    EditionableWorldwideOrganisation ||--|{ Attachment : "has many"
+
+    EditionWorldLocation ||--|| WorldLocation : "has one"
+    EditionOrganisation ||--|| Organisation : "has one"
+    EditionRole ||--|| Role : "has one"
+    EditionEditionableWorldwideOrganisation ||--|| WorldNewsStory : "has one"
+```
+
+## Proposal
+
+This proposal is that we shall apply the edition workflow to all the remaining non-editionable documents in Whitehall Publisher (and all their associated documents):
+
+* Fields of Operation
+* Governments
+* Groups
+* Organisations
+* People
+* Role Appointments
+* Roles
+* Take part pages
+* Topical Events
+* World Location News
+
+This would be done in the same way as Worldwide Organisations: a new editionable model (and associated models) would be introduced alongside the existing non-editionable model, but its usage kept behind a feature flag. We would then create a migration to move the data from one model to the other, whilst also flipping the feature flag at the same time.
+
+Furthermore we propose that new document types must be editionable. If a new document type needs to be added to any publishing application, it must always follow an edition workflow.
```

Comment:
So I'd suggest adding a `Blocked` section to the bottom here, to cover Kevin and Ryan's concerns that the code isn't factored appropriately to provide a good foundation for making more things editionable.

That said, some things (like "new document types must be editionable") do seem like actual decisions we could make. I guess we could split out two RFCs (1 - new document types must be editionable, 2 - make non-editionable document types editionable), but that's a bit of a faff... Maybe "partially blocked"?

Reply:
I'll stick with this being "partially blocked", but not dictate that future document types must be editionable from now.  This is because a new complex document type may hit the same tech debt problems.  I'll suggest that the new document should be editionable from the point that the underlying tech debt is fixed.